### DISCUSSION TITLE
Gameplay cursor for touch input is delay

### DISCUSSION THREAD

**Discussion by Shirou27** [ppy/osu#19565] [Q&A] [answered]
The gameplay cursor for touch input is delay while playing or clicking circles, because of this it makes me feel like the touch response is delay even though its not.

My phone spec if you need it:
Model: Redmi Note 9
Refresh Rate: 60Hz
Chipset: MediaTek Helio G85 (12nm)
CPU: Octa-core (2x2.0 GHz Cortex-A75 & 6x1.8 GHz Cortex-A55)
GPU: Mali-G52 MC2

i don't know if my recorded video can make you see it(btw, the recorded video is in 30fps to avoid serious lagging while playing)

https://user-images.githubusercontent.com/105548541/182584932-086ca16d-8366-4ac3-85d0-ad6f22c8c75e.mp4

**Comment by peppy** [comment, marked as answer]
The cursor is showing the actual position of input. There's no way it can be delayed in any way. I can't see what you mean from your video, is there a specific frame where you are expecting different behaviour that you can point out?

**Reply by Shirou27** [reply]
I slow down the video myself and there's really no delay but it feels different when I'm playing. i really feel a delay in the cursor. If you didn't see anything wrong then its probably just my imagination

Sorry for the inconvenience🙏

**Reply by peppy** [reply]
This is one of the reasons we left the cursor hidden until now. I'd recommend you hide it.

**Reply by Theighlin** [reply]
I'm thinking most phones nowadays have a higher touch sampling rate than the refresh rate, so it's a real fact that there is some gathered input that can't be displayed, as for how noticeable it is, i can't really say. (Though it seems his phone model has both the refresh rate and touch sampling rate at 60hz)
